pdf/model/fonts/ttfparser.go

```diff
@@ -258,117 +369,396 @@ func (t *ttfParser) ParseCmap() (err error) {
+	"bar", "braceright", "asciitilde", "Adieresis", "Aring",
+	"Ccedilla", "Eacute", "Ntilde", "Odieresis", "Udieresis", "aacute",
+	"agrave", "acircumflex", "adieresis", "atilde", "aring",
+	"ccedilla", "eacute", "egrave", "ecircumflex", "edieresis",
+	"iacute", "igrave", "icircumflex", "idieresis", "ntilde", "oacute",
+	"ograve", "ocircumflex", "odieresis", "otilde", "uacute", "ugrave",
+	"ucircumflex", "udieresis", "dagger", "degree", "cent", "sterling",
+	"section", "bullet", "paragraph", "germandbls", "registered",
+	"copyright", "trademark", "acute", "dieresis", "notequal", "AE",
+	"Oslash", "infinity", "plusminus", "lessequal", "greaterequal",
+	"yen", "mu", "partialdiff", "summation", "product", "pi",
+	"integral", "ordfeminine", "ordmasculine", "Omega", "ae", "oslash",
+	"questiondown", "exclamdown", "logicalnot", "radical", "florin",
+	"approxequal", "Delta", "guillemotleft", "guillemotright",
+	"ellipsis", "nonbreakingspace", "Agrave", "Atilde", "Otilde", "OE",
+	"oe", "endash", "emdash", "quotedblleft", "quotedblright",
+	"quoteleft", "quoteright", "divide", "lozenge", "ydieresis",
+	"Ydieresis", "fraction", "currency", "guilsinglleft",
+	"guilsinglright", "fi", "fl", "daggerdbl", "periodcentered",
+	"quotesinglbase", "quotedblbase", "perthousand", "Acircumflex",
+	"Ecircumflex", "Aacute", "Edieresis", "Egrave", "Iacute",
+	"Icircumflex", "Idieresis", "Igrave", "Oacute", "Ocircumflex",
+	"apple", "Ograve", "Uacute", "Ucircumflex", "Ugrave", "dotlessi",
+	"circumflex", "tilde", "macron", "breve", "dotaccent", "ring",
+	"cedilla", "hungarumlaut", "ogonek", "caron", "Lslash", "lslash",
+	"Scaron", "scaron", "Zcaron", "zcaron", "brokenbar", "Eth", "eth",
+	"Yacute", "yacute", "Thorn", "thorn", "minus", "multiply",
+	"onesuperior", "twosuperior", "threesuperior", "onehalf",
+	"onequarter", "threequarters", "franc", "Gbreve", "gbreve",
+	"Idotaccent", "Scedilla", "scedilla", "Cacute", "cacute", "Ccaron",
+	"ccaron", "dcroat",
+}
+
+// Seek moves the file pointer to the table named `tag`.
+func (t *ttfParser) Seek(tag string) error {
 	ofs, ok := t.tables[tag]
-	if ok {
-		t.f.Seek(int64(ofs), os.SEEK_SET)
-	} else {
-		err = fmt.Errorf("table not found: %s", tag)
+	if !ok {
+		return fmt.Errorf("table not found: %s", tag)
 	}
-	return
+	t.f.Seek(int64(ofs), os.SEEK_SET)
+	return nil
 }
 
+// Skip moves the file point n bytes forward.
 func (t *ttfParser) Skip(n int) {
 	t.f.Seek(int64(n), os.SEEK_CUR)
 }
 
+// ReadStr reads `length` bytes from the file and returns them as a string, or an error if there was
+// a problem.
 func (t *ttfParser) ReadStr(length int) (str string, err error) {
 	var n int
 	buf := make([]byte, length)
 	n, err = t.f.Read(buf)
-	if err == nil {
-		if n == length {
-			str = string(buf)
-		} else {
-			err = fmt.Errorf("unable to read %d bytes", length)
-		}
+	if err != nil {
+		return
 	}
+	if n == length {
+		str = string(buf)
+	} else {
+		err = fmt.Errorf("unable to read %d bytes", length)
+	}
+	return
```

Comment:
This file needs to undergo some revision and refactoring with respect to formatting at some point.
Full of naked returns and different code style.

Can leave for now.